### PR TITLE
control scan and download only by id 

### PR DESCRIPTION
**Pull request by kooomix**
## Describe your changes
- download control only by id.
- scan control only by id

## Checklist before requesting a review

- [x] My code follows the style guidelines of this project
- [x] I have commented on my code, particularly in hard-to-understand areas
- [x] I have performed a self-review of my code
- [x] If it is a core feature, I have added thorough tests.
- [x] New and existing unit tests pass locally with my changes

**Please open the PR against the `dev` branch (Unless the PR contains only documentation changes)**

### CODE REVIEW

**Comment by dwertent** [comment]
Waiting for approval from the product.
@Bezbran

File: core/meta/datastructures/v1/download.go
Comment (dwertent) [inline core/meta/datastructures/v1/download.go:10]: The `name` means identifier.
There is no need to add a new field.
If this is confusing, replace the `name` filed with `ID`/`identifier`

File: core/cautils/scaninfo.go
Comment (dwertent) [inline core/cautils/scaninfo.go:219]: please see the comment I added to the structure

File: core/cautils/scaninfo.go
Comment (dwertent) [inline core/cautils/scaninfo.go:97]: This is the identifier. It does not make sense to have a name and ID and to add a comment regarding the types of identifiers.
You either create new structs for identifying controls/frameworks or that you change the `name` to `ID`/`identifier`

**Review by dwertent — CHANGES_REQUESTED** [review]
I will expect to see a validator function regarding the controlID.
If the input control ID is invalid, we should return an error saying we do not support controls names as identifiers and that the user should use control ID instead `kubescape list controls`.

**Comment by kooomix** [comment]
> I will expect to see a validator function regarding the controlID. If the input control ID is invalid, we should return an error saying we do not support controls names as identifiers and that the user should use control ID instead `kubescape list controls`.

Are we sure we need a special validation function? In case a control wasn't found we can just write back something like:
"control id 'someInvalidID' wasn't found. Make sure to use a valid control id - `kubescape list controls`"

File: core/meta/datastructures/v1/download.go
Comment (kooomix) [reply core/meta/datastructures/v1/download.go]: Done

File: core/cautils/scaninfo.go
Comment (kooomix) [reply core/cautils/scaninfo.go]: Done

File: core/cautils/scaninfo.go
Comment (kooomix) [reply core/cautils/scaninfo.go]: Done

**Comment by dwertent** [comment]
> Are we sure we need a special validation function? In case a control wasn't found we can just write back something like:
"control id 'someInvalidID' wasn't found. Make sure to use a valid control id - kubescape list controls"

@kooomix  Because we are changing the behavior here, we should **explicitly** mention that we do not support control names for input.

File: smoke_testing/test_scan.py
Comment (dwertent) [inline smoke_testing/test_scan.py:51]: Please remove this

**Comment by kooomix** [comment]
Than I can just mention we don't support control names in case control was not found. Without needing to guess if the user tried a name or not.

**Comment by kooomix** [comment]
> > Are we sure we need a special validation function? In case a control wasn't found we can just write back something like:
> > "control id 'someInvalidID' wasn't found. Make sure to use a valid control id - kubescape list controls"
> 
> @kooomix Because we are changing the behavior here, we should **explicitly** mention that we do not support control names for input.

Than we can just mention we don't support control names in case control was not found. Without needing to guess if the user tried a name or not.

**Review by dwertent — APPROVED** [review]

**Pull request merged**
